model/src/main/java/com/cloudera/nav/sdk/model/entities/HdfsEntity.java

```diff
@@ -48,13 +50,19 @@ public HdfsEntity(String id) {
     setIdentity(id);
   }
 
+  /**
+   * Either the entity id must be present or the file system path must be
+   * present for an HdfsEntity to be valid. The source id must also be
+   * present for the HdfsEntity to be valid.
+   */
   @Override
   public void validateEntity() {
-    if (Strings.isNullOrEmpty(this.getIdentity()) &&
-        Strings.isNullOrEmpty(this.getFileSystemPath())) {
+    if ((Strings.isNullOrEmpty(this.getIdentity()) &&
+        Strings.isNullOrEmpty(this.getFileSystemPath())) ||
+        Strings.isNullOrEmpty(this.getSourceId())) {
       throw new IllegalArgumentException(
           "Either the Entity Id or file system path used" +
-              " to generate the id must be present");
+              " to generate the id must be present along with the source id");
```

Comment:
If you've already have the ID, why do you need to generate the id?

Reply:
The user might not know the id, in which the case the user passes in the filesystem path and the source id and the server knows how to generate the id for the entity.